### PR TITLE
Added Itch manifest & launch configs

### PR DESCRIPTION
**Pull request by aaronfranke**
Part of solving https://github.com/OpenRA/OpenRA/issues/14214

* `.itch.toml` is the main manifest file. This is the main thing I'm adding. It contains information about how to launch the game. The `os` property currently does nothing, but in the future (https://github.com/itchio/itch/pull/1605), it will make only the relevant options for your operating system show up.

* ~~`.itch/isolate-app.profile` is required for the Itch sandbox.~~ Ignore the entire `.itch` folder as Butler ignores it and it should be treated as temporary files.

* `mono` ~~and `mono-macos` are~~ is a symlink~~s~~ to Mono on Linux ~~and MacOS~~. The Itch app sets the working directory to the location of the executable, so directly linking to Mono in `.itch.toml` is not an option. The solution is simply to create symlink~~s~~ to `mono` in the OpenRA directory, as I've done.

* These files do not cause any problems when launching OpenRA without the Itch app.

I have not actually tested this on MacOS, since I do not own a Mac. I got the Mono path from [StackOverflow](https://stackoverflow.com/questions/32542535/how-to-install-mono-on-macos-so-mono-works-in-terminal). However, I have high confidence it will work, since it's a similar method to launching on Linux.

### CODE REVIEW

**Comment by pchote** [comment]
I’m not sure how the itch side of things works, but from the OpenRA side launching OpenRA.Game.exe directly is a bad idea.  This bypasses all the platform specific shims in their respective launchers, which means no crash handing, no application icons and incorrect names, pinning to taskbar/dock/launcher leaves broken shortcuts that crash on launch, plus a few smaller issues.  Can you please make this run the platform launchers instead?  You should then be able to drop the mono symlinks.

File: .itch.toml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+[[actions]]
+os = "windows"
+name = "Tiberian Dawn - Windows"
+path = "OpenRA.Game.exe"
+args = ["Game.Mod=cnc"]
+sandbox = true
+
+[[actions]]
+os = "windows"
+name = "Tiberian Sun - Windows"
```
Comment (pchote) [inline .itch.toml]: Can you please remove these? Tiberian Sun is not ready for a public release.

**Comment by aaronfranke** [comment]
Can you clarify, what are the platform launchers?

**Comment by pchote** [comment]
The .app bundles on macOS, `/usr/(bin|games)/openra-(cnc|d2k|ra)` on linux, and `(RedAlert|TiberianDawn|Dune2000).exe` on windows.  They are generated as part of the package creation scripts.

**Comment by aaronfranke** [comment]
For Windows launchers: Sure!

> `/usr/(bin|games)/openra-(cnc|d2k|ra)` on linux

No, I can't just run these, not directly. These files existing depend on OpenRA being installed through the system's package manager, and they point to `/usr/lib/openra`. Releasing OpenRA builds through Itch means that you launch the version that downloads to the Itch app directory (`~/.config/itch/apps/OpenRA` on Linux and the equivalent paths on Windows and Mac).

What I can do is copy-paste these launch scripts, place them in the main OpenRA directory, and commit them along with this patch. However, you mentioned that they are generated, so I am wondering if copy-pasting them is the wrong approach.

Would the MacOS .app bundles be included within the game folder like on Windows?

**Comment by pchote** [comment]
The app bundles wrap the game directory plus other macOS specific files into a self contained portable application.  Think of them like the Linux .appimage format, which was modeled after them.

**Comment by aaronfranke** [comment]
Itch doesn't seem to like it when I pass it a Linux shell script directly. https://github.com/itchio/itch/issues/1608

Windows and Mac now use launchers. The concept of pinning doesn't exist on many Linux DEs, and almost certainly not for pinning the launcher script with an icon. I've also experimented with calling `.desktop` files, which has issues without absolute paths. Launching on Linux the way it's currently set up with `mono` etc is probably the best way to go about it, as it's consistent and robust.

**Comment by pchote** [comment]
How does itch get the game files for these entries to launch?

**Comment by pchote** [comment]
Even without pinning, launching the Game.exe directly will still omit the icons and titles from the launcher and window switch menus, or am I missing something?

**Comment by aaronfranke** [comment]
In the future (https://github.com/OpenRA/OpenRA/issues/11352) we can set up Itch's `butler` utility to automatically upload built game files. We would add this to the build scripts. Until then, we can upload ZIP files through a web browser.

openra

On my system, this is what OpenRA looks like, when launched from the shell script and `/usr/share/applications/openra-ra.desktop` installed with the .deb package from the website, and also when launched with `mono OpenRA.Game.exe Game.Mod=ra` from Itch. I'm pretty sure the icon on Linux depends on the executable and not the launch method.

**Comment by pchote** [comment]
Bottom line, `mono OpenRA.Game.exe Game.Mod=ra` is just as icon-y as it currently is with the `.deb`.

This isn't how we expect people to run OpenRA from the deb!

**Comment by aaronfranke** [comment]
At least in my desktop environment, launching with the `.desktop` file doesn't add an icon to the taskbar.

**Comment by aaronfranke** [comment]
I should clarify, I did not mean to say that you expect users to launch with `mono OpenRA.Game.exe` when installed with the deb. I know the `.desktop` tied to the script is intended. I am just saying that once Itch launches this way, vs now, the results are identical.

Anyway, basically the only error handling the script really does, on Linux at least, is say "I crashed, want to view help?". I already set up the manual/wiki as a launch option. So really the only reason we'd want to use the existing launch systems is for the icon, which the script doesn't do...

**Comment by aaronfranke** [comment]
I believe this is merge-able as-is. We can make tweaks later, if necessary.

**Comment by pchote** [comment]
Can you please outline the steps to get OpenRA with the metadata file imported and running inside itch?  This will have important implications on where the files are supposed to live and how they interact with the rest of our code - having this manifest in the root of the source code dir is probably not correct.

I have been assuming that this is going to need its own packaging "platform" to generate a (or ideally one per platform) package that could be consumed by the itch backend.  Am I missing something?

**Comment by aaronfranke** [comment]
Eventually, when we set up [Itch.io butler](https://github.com/OpenRA/OpenRA/issues/11352) in some part of the build process:

1. The files to be uploaded to Itch, for each platform, go to their own folder. Not a zip.

2. We push these to Itch using their butler utility.

We definitely could do something where these files are placed in a subfolder in `packaging`, then copied to the main game directory when generating and uploading Itch builds. See latest commit.

**Comment by aaronfranke** [comment]
Do not merge for now. The creator of Itch is looking into better methods of handling Mono games, dependencies, and launching mechanisms.

https://github.com/itchio/itch/issues/1606

https://github.com/itchio/itch/issues/1642

**Comment by pchote** [comment]
@aaronfranke does the new AppImage packaging in #15086 make this any easier for you?
We could potentially also port over the `-winportable` packaging from the Mod SDK (basically just a zip file with the contents of what would be installed to Program Files), which would then provide "portable" bundles for all platforms that Itch could download and chain through to.

**Comment by aaronfranke** [comment]
@pchote It very likely will. We still need Mono installed, so we still need Itch to add Mono to their list of prerequisites for best compatibility, but we should be able to fix everything on OpenRA's end for users that have Mono manually installed.

I'll look into it sometime this week.

**Comment by pchote** [comment]
Any progress here @aaronfranke? Shall we close this PR until the itch-side dependencies are sorted out?

**Comment by aaronfranke** [comment]
Yeah, it'd be best to wait and see what Itch does in regards to Mono support. Closing, for now.

**Pull request closed without merging**